src/Core/CoreKernel.php

```diff
@@ -87,31 +87,40 @@ protected function bootDatabaseGlobals()
         global $databaseConfig;
         global $database;
 
-        // Case 1: $databaseConfig global exists. Merge $database in as needed
-        if (!empty($databaseConfig)) {
-            if (!empty($database)) {
-                $databaseConfig['database'] =  $this->getDatabasePrefix() . $database . $this->getDatabaseSuffix();
+        for ($i = 0; $i <= 99; $i++) {
+            if ($i === 0) {
+                $key = 'default';
+            } else {
+                $key = DB::getReplicaConfigKey($i);
+                if (!DB::hasConfig($key)) {
+                    break;
+                }
             }
 
-            // Only set it if its valid, otherwise ignore $databaseConfig entirely
-            if (!empty($databaseConfig['database'])) {
-                DB::setConfig($databaseConfig);
-
-                return;
+            // Case 1: $databaseConfig global exists. Merge $database in as needed
+            if (!empty($databaseConfig)) {
+                if (!empty($database)) {
+                    $databaseConfig['database'] =  $this->getDatabasePrefix() . $database . $this->getDatabaseSuffix();
+                }
```

Comment:
Won't this override the primary `$databaseConfig['database']` value with whatever the highest replica is that exists? Or if this is the same value every time, shouldn't we only do this once?

Same with case 2 below.

Reply:
I don't this so? `$database` is a global to specify use the `SS_mysite` database or equivalent. CoreKernel will set it to Environment::getEnv('SS_DATABASE_NAME').  In the docs I've already mentioned that replicas must have the same database name as the primary

Reply:
The relevant bit of code can be boiled down to:

```php
for ($i = 0; $i <= DB::MAX_REPLICAS; $i++) {
    if (!empty($databaseConfig) && !empty($database)) {
        $databaseConfig['database'] =  $this->getDatabasePrefix() . $database . $this->getDatabaseSuffix();
    }
}
```

Which means we're setting this same value up to 100 times.

Reply:
Updated